hw7/Sem07Task34.py

```diff
@@ -0,0 +1,23 @@
+# Задача 34: Винни-Пух попросил Вас посмотреть, есть ли в его стихах ритм. Поскольку
+# разобраться в его кричалках не настолько просто, насколько легко он их придумывает, Вам
+# стоит написать программу. Винни-Пух считает, что ритм есть, если число слогов (т.е. число
+# гласных букв) в каждой фразе стихотворения одинаковое. Фраза может состоять из одного
+# слова, если во фразе несколько слов, то они разделяются дефисами. Фразы отделяются друг
+# от друга пробелами. Стихотворение Винни-Пух вбивает в программу с клавиатуры. В ответе
+# напишите “Парам пам-пам”, если с ритмом все в порядке и “Пам парам”, если с ритмом все не
+# в порядке
+# Ввод: 
+# пара-ра-рам рам-пам-папам па-ра-па-дам
+# Вывод:
+# Парам пам-пам
+
+
+def CheckRythm(song):
+    phrases = song.lower().split() # разбиваем кричалку на фразы
+    f = lambda x: sum(1 for i in x if i in 'аеёиоуыэюя') # лямбда функция подсчета гласных
+    tmp = f(phrases[0]) # считаем количество слогов в первой фразе
+    if all([f(i) == tmp for i in phrases]): # если количество слогов во всех фразах, равно количеству слогов в первоф фразе, то:
```

Comment:
Можно просто передать генератор
```suggestion
    if all(f(i) == tmp for i in phrases): # если количество слогов во всех фразах, равно количеству слогов в первоф фразе, то:
```